src/globus_cli/commands/timer/update.py

```diff
@@ -0,0 +1,56 @@
+import datetime
+from typing import Any, Dict, Optional
+
+import click
+
+from globus_cli.login_manager import LoginManager
+from globus_cli.parsing import command
+from globus_cli.termio import FORMAT_TEXT_RECORD, formatted_print
+
+from ._common import DATETIME_FORMATS, JOB_FORMAT_FIELDS
+
+
+@command("update", short_help="Modify a Timer job", hidden=True)
+@LoginManager.requires_login(LoginManager.TIMER_RS)
+@click.argument("JOB_ID")
+@click.option(
+    "--name",
+    type=str,
+    help="Name to identify this job in the timer service (not necessarily unique)",
+)
+@click.option(
+    "--stop-after-date",
+    type=click.DateTime(formats=DATETIME_FORMATS),
+    help="Stop running the transfer after this date",
+)
+@click.option(
+    "--stop-after-runs",
+    type=int,
+    help="Stop running the transfer after this number of runs have happened",
+)
+def update_command(
+    login_manager: LoginManager,
+    job_id: str,
+    name: Optional[str],
+    stop_after_date: Optional[datetime.datetime],
+    stop_after_runs: Optional[int],
+):
+    """
+    Update certain parameters on an existing Timer job given its UUID.
+
+    Start time and parameters such as transfer options cannot be changed once a job is
+    submitted—only the name, interval, and stop conditions.
+    """
+    timer_client = login_manager.get_timer_client()
+    data: Dict[str, Any] = {}
+    if name:
+        data["name"] = name
+    if stop_after_date:
+        data["stop_after"] = {"date": stop_after_date.isoformat()}
+    if stop_after_runs:
+        data["stop_after"] = {**data.get("stop_after", {}), "n_runs": stop_after_runs}
```

Comment:
This feels awkward, but I get it... How do you feel about `defaultdict(dict)`? Better, worse, or about the same?

Another option, to the tune of "super verbose, but super clear":

```suggestion
    if stop_after_date or stop_after_runs:
        data["stop_after"] = {}
        if stop_after_date:
            data["stop_after"]["date"] = stop_after_date.isoformat()
        if stop_after_runs:
            data["stop_after"]["n_runs"] = stop_after_runs
```

Reply:
The suggestion is much clearer at the expense of 2 lines, I'll definitely go with that.